src/common/tools/gen_listings_and_webworkers.ts

```diff
@@ -0,0 +1,90 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as process from 'process';
+
+import { crawl } from './crawl.js';
+
+function usage(rc: number): void {
+  console.error(`Usage: tools/gen_listings [options] [OUT_DIR] [SUITE_DIRS...]
+
+For each suite in SUITE_DIRS, generate listings into OUT_DIR/{suite}/listing.js,
+and generate Web Worker proxies in OUT_DIR/{suite}/webworker/**/*.worker.js for
+every .spec.js file. (Note {suite}/webworker/ is reserved for this purpose.)
+
+Example:
+  tools/gen_listings gen/ src/unittests/ src/webgpu/
```

Comment:
oops, I missed something here
```suggestion
  tools/gen_listings_and_webworkers gen/ src/unittests/ src/webgpu/
```

Reply:
Done